Packs/Azure/Integrations/Azure/Azure.py

```diff
@@ -1934,6 +1938,218 @@ def cosmosdb_update_command(client: AzureClient, params: dict[str, Any], args: D
     )
 
 
+def azure_billing_usage_list_command(client: AzureClient, params: dict, args: dict) -> CommandResults:
+    """
+    Retrieves actual usage and cost details from Azure Consumption API.
+
+    This command provides detailed billing usage information for Azure resources over a specified time period.
+    It supports filtering by various criteria and includes pagination for large datasets. The command returns
+    usage quantities, costs, and resource details for comprehensive billing analysis.
+
+    Args:
+        client (AzureClient): Azure client instance for API communication
+        params (dict): Configuration parameters from integration settings
+        args (dict): Command arguments containing:
+            - subscription_id: Azure subscription ID (required)
+            - expand_result: Expand usage details with additional properties
+            - filter: OData filter expression for filtering results
+            - metric: Specific metric to retrieve (e.g., ActualCost, UsageQuantity)
+            - max_results: Maximum number of results to return (default: 50)
+            - next_page_token: Token for pagination
+
+    Returns:
+        CommandResults: Contains usage data with costs, quantities, and time periods,
+                      including pagination support via next page tokens
+
+    Raises:
+        DemistoException: If Azure API call fails, subscription not found, or invalid parameters provided
+    """
+    subscription_id = get_from_args_or_params(params=params, args=args, key="subscription_id")
+    expand = args.get("expand_result")
+    filter_ = args.get("filter")
+    metric = args.get("metric")
+    max_results = int(args.get("max_results", 50))
+    next_page_token = args.get("next_page_token")
+
+    scope = f"/{subscription_id}"
+    url = f"{scope}/providers/Microsoft.Consumption/usageDetails"
+    api_version = "2021-10-01"
+    params_ = {}
+    if expand:
+        params_["$expand"] = expand
+    if filter_:
+        params_["$filter"] = filter_
+    if metric:
+        params_["metric"] = metric.lower().replace(" ", "")
+    params_["api-version"] = api_version
+    params_["$top"] = max_results
```

Comment:
We can build the parmas dict regardless of those ifs and the remove_empty_elements .